src/rvw.cpp

```diff
@@ -146,23 +192,57 @@ void vwtrain(Rcpp::List vwmodel, std::string data_path="", Rcpp::Nullable<Rcpp::
 //'@param probs_path Path to file where to save predictions
 //'@param readable_model Print trained model in human readable format ("hashed") 
 //'and also with human readable features ("inverted")
-//'@param quiet Do not print anything to the console 
+//'@param quiet Do not print anything to the console
+//'@param namespaces [list or yaml file] For \code{df2vw}. Name of each namespace and
+//'  each variable for each namespace can be a R list, or a YAML
+//'  file example namespace with the IRIS database: namespaces =
+//'  list(sepal = list('Sepal.Length', 'Sepal.Width'), petal = list('Petal.Length',
+//'  'Petal.Width') this creates 2 namespaces (sepal
+//'  and petal) containing the features defined by elements of this lists.
+//'@param keep_space [string vector] For \code{df2vw}. Keep spaces for this features
+//'Example:"FERRARI 4Si"
+//'With \code{keep_space} will be "FERRARI 4Si" and will be treated as two features
+//'Without \code{keep_space} will be "FERRARI_4Si" and will be treated as one feature
+//'@param targets [string or string vector] For \code{df2vw}.
+//'If \code{[string]} then will be treated as vector with real number labels for regular VW input format. 
+//'If \code{[string vector]} then will be treated as vectors with class costs for wap and csoaa 
+//'multi-class classification algorithms or as vectors with actions for Contextual Bandit algorithm. 
+//'@param probabilities [string vector] For \code{df2vw}. Vectors with action probabilities for Contextual Bandit algorithm.
+//'@param weight [string] For \code{df2vw}. Weight (importance) of each line of the dataset.
+//'@param base [string] For \code{df2vw}. Base of each line of the dataset. Used for residual regression.
+//'@param tag [string] For \code{df2vw}. Tag of each line of the dataset.
+//'@param multiline [integer] number of labels (separate lines) for multilines examle
 //'@return Numerical vector containing predictions
+//'@import tools
 //'@examples
 //'ext_train_data <- system.file("extdata", "binary_train.vw", package = "rvwgsoc")
 //'ext_test_data <- system.file("extdata", "binary_valid.vw", package = "rvwgsoc") 
 //'test_vwmodel <- vwsetup()
-//'vwtrain(test_vwmodel, data_path = ext_train_data)
-//'vwtrain(test_vwmodel, data_path = ext_test_data)
+//'vwtrain(test_vwmodel, data = ext_train_data)
+//'vwtrain(test_vwmodel, data = ext_test_data)
 // [[Rcpp::export]]
-Rcpp::NumericVector vwtest(Rcpp::List vwmodel, std::string data_path="", std::string probs_path = "", Rcpp::Nullable<Rcpp::String> readable_model=R_NilValue, bool quiet=false) {
-    // if (!Rcpp::mod.inherits("vw")) Rcpp::stop("Input model is not VW model");
-    // Rcpp::Rcout << vwmodel.attr("class") << std::endl;
-    std::string data_str = check_data(vwmodel, data_path, "test");
+Rcpp::NumericVector vwtest(Rcpp::List & vwmodel, SEXP & data=R_NilValue, std::string probs_path = "", Rcpp::Nullable<Rcpp::String> readable_model=R_NilValue, bool quiet=false,
```

Comment:
Note the `Nullable` definition in the parameters given here... But, this was later dropped infavor of `SEXP &`. Why?

Reply:
Seems like I really misunderstood this logic, but I will rework it.